components/server/src/bitbucket-server/bitbucket-server-api.ts

```diff
@@ -4,15 +4,22 @@
  * See License.AGPL.txt in the project root for license information.
  */
 
-import fetch from "node-fetch";
+import fetch, { RequestInfo, RequestInit, Response } from "node-fetch";
 import { User } from "@gitpod/gitpod-protocol";
 import { inject, injectable } from "inversify";
 import { AuthProviderParams } from "../auth/auth-provider";
 import { BitbucketServerTokenHelper } from "./bitbucket-server-token-handler";
 import { CancellationToken } from "vscode-jsonrpc";
 import { URLSearchParams } from "url";
 import * as qs from "node:querystring";
+import { log } from "@gitpod/gitpod-protocol/lib/util/logging";
 
+async function loggedFetch(url: RequestInfo, options: RequestInit): Promise<Response> {
+    const before = Date.now();
+    const result = await fetch(url, options);
+    log.info(`bitbucket fetch`, { url: url, time: `${Date.now() - before} ms` });
```

Comment:
:bubbles: While this works to some degree, a metric would be a bit easier to read.

Reply:
I did it like this in order to see what calls are made per resolveContext request.